### PR TITLE
X86: fix kernel image name to 'vmlinux'

### PR DESCRIPTION
**Pull request by gkkpch**
I suggest this one to replace #473

### CODE REVIEW

File: recipes/devices/families/x86.sh [hunk 1]
```diff
@@ -160,17 +160,14 @@ device_chroot_tweaks_pre() {
...
-
-  ls -l /boot
+  log "Chanege linux kernel image name to 'vmlinuz'"
+
+  # Rename linux kernel to a fixed name, like we do for any other platform.
+  # We only have one and we should not start multiple versions.
+  # - our OTA update can't currently handle that and it blows up size of /boot and /lib.
+
+  # This rename is safe, because we have only one vmlinuz in /boot
+  mv $(ls /boot/vmlinuz*) /boot/vmlinuz
```
Comment (ashthespy) [inline recipes/devices/families/x86.sh:170]: Shouldn't this just be `mv /boot/vmlinuz* /boot/vmlinuz`?

**Comment by ashthespy** [comment]
Side note, it would be nice to add an explicit log line with the actual version of the kernel installed. Makes it easier to later check logs..

File: recipes/devices/families/x86.sh [hunk 2]
```diff
@@ -246,7 +243,7 @@ EOF
   sed -i "s/%%IMGPART%%/${UUID_IMG}/g" ${grub_cfg}
   sed -i "s/%%BOOTPART%%/${UUID_BOOT}/g" ${grub_cfg}
   sed -i "s/%%DATAPART%%/${UUID_DATA}/g" ${grub_cfg}
-  sed -i "s/%%KRNL%%/${KRNL}/g" ${grub_cfg}
+
```
Comment (ashthespy) [inline recipes/devices/families/x86.sh:246]: Wouldn't this lead to a nonbootable image -- [`grub.cfg`](https://github.com/volumio/platform-x86/blob/0e4dd9c56ad0c6224d7854685abdf03f454b2278/packages-buster/efi/BOOT/grub.cfg#L117) will try and load `%%KRNL%%` as the kernel.

**Comment by gkkpch** [comment]
It already does that with the dpkg install and also shows the version number when building the initramfs.

File: recipes/devices/families/x86.sh (hunk 1 above)
Comment (gkkpch) [reply recipes/devices/families/x86.sh:170]: even simpler when that works, never thought of that

File: recipes/devices/families/x86.sh (hunk 2 above)
Comment (ashthespy) [reply recipes/devices/families/x86.sh:246]: Ah, https://github.com/volumio/platform-x86/commit/e9aad47c415c73b772e8542d63f02e1720130bae covers that! 👍

Comment (gkkpch) [reply recipes/devices/families/x86.sh:246]: > Wouldn't this lead to a nonbootable image -- [`grub.cfg`](https://github.com/volumio/platform-x86/blob/0e4dd9c56ad0c6224d7854685abdf03f454b2278/packages-buster/efi/BOOT/grub.cfg#L117) will try and load `%%KRNL%%` as the kernel.

Yes, but the platform files were updated, this works fine

**Comment by gkkpch** [comment]
Do we agree to merge this one?
(I know, the comments have a typo, I'll fix that later)

**Comment by volumio** [comment]
@gkkpch if you confirm that all is ok and tested, I'll merge

**Comment by volumio** [comment]
@gkkpch please confirm when I can merge (after its been tested)

**Comment by gkkpch** [comment]
Yes, this was tested before pushing.

**Comment by volumio** [comment]
Thanks!

**Pull request merged**